appeals/web/src/server/lib/postcode.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * @param {string|null|undefined} str
+ * @returns {boolean}
+ */
+export function stringIsValidPostcodeFormat(str) {
+	if (str !== null && str !== undefined) {
+		const regex = /^([A-Za-z]{1,2}\d[A-Za-z\d]? ?\d[A-Za-z]{2}|GIR ?0A{2})$/gm;
```

Comment:
Hey @james-hill-kc this looks good but probably we should consolidate, as the API also validate the postcode through: https://github.com/Planning-Inspectorate/appeals-back-office/blob/91ad0224ab5e88f85d532ffff30d8934e8fa7172/appeals/api/src/server/endpoints/neighbouring-sites/neighbouring-sites.validators.js#L13

Reply:
Good spot, this has been updated to use the same regex. If there's a way of sharing this from some common data source please let me know, as that would be preferable :)